locust/test/test_main.py

```diff
@@ -1475,11 +1818,29 @@ def test_expect_workers(self):
                 stderr=PIPE,
                 text=True,
             )
-            _, stderr = proc.communicate()
-            self.assertIn("Waiting for workers to be ready, 0 of 2 connected", stderr)
-            self.assertIn("Gave up waiting for workers to connect", stderr)
-            self.assertNotIn("Traceback", stderr)
-            self.assertEqual(1, proc.returncode)
+
+            def check_output():
```

Comment:
This function doesnt look very useful to me, and it is only used once. Its probably easier to read if you just call `proc.stderr.readline().strip()` directly